### PR TITLE
Workoff workers exit if not on latest task definition

### PR DESCRIPTION
**Pull request by yakloinsteak**

### CODE REVIEW

File: app/jobs/worker_status.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    Rails.logger.warn "Exiting because I am a workoff worker that isn't on the latest version"
+    exit!(0)
+  end
+
+  private
+
+  def latest_deployment_is_at_least_partially_finished
+    return true if Rails.env.development? || Rails.env.test?
+
+    # Couldn't get Curl.get to work
```
Comment (eanders) [inline app/jobs/worker_status.rb:65]: did you `require 'curb'`?

Comment (yakloinsteak) [reply app/jobs/worker_status.rb:65]: I didn't, but the error wasn't that the class didn't exist. That's an odd way for it to fail if that's all I needed. Want me to try it out?

Comment (eanders) [reply app/jobs/worker_status.rb:65]: Let's leave it for now.

**Comment by yakloinsteak** [comment]
Updated this to be a before hook instead of an after hook and tested on QA. Seems to work, and I cleaned up the extra "filler" jobs I was testing with.

**Comment by yakloinsteak** [comment]
Don't merge yet. I need to retest with the unlock job part wired in.

**Review by eanders — APPROVED** [review]

**Pull request merged**